components/dashboard/sections/collectives/FreezeAccountModal.tsx

```diff
@@ -0,0 +1,410 @@
+  const intl = useIntl();
+  const { toast } = useToast();
+  const [useMessageForAccountAdmins, setUseMessageForAccountAdmins] = useState(false);
+  const [messageForAccountAdmins, _setMessageForAccountAdmins] = useState('');
+  const [pauseContributions, setPauseContributions] = useState(true);
+  const [hasConfirmed, setHasConfirmed] = useState(false);
+  const [submitAction, setSubmitAction] = useState<ModalState | null>(null);
+  const { data, loading, error } = useQuery(freezeAccountModalQuery, {
+    variables: { accountId: collective.id },
+    context: API_V2_CONTEXT,
+  });
+  const [editAccountFreezeStatus, { loading: submitting }] = useMutation(editAccountFreezeStatusMutation, {
+    context: API_V2_CONTEXT,
+  });
+  const setMessageForAccountAdminsFromInputEvent = React.useCallback(
+    e => _setMessageForAccountAdmins(e.target.value),
+    [],
+  );
+
+  const defaultState = collective.isFrozen ? 'UNFREEZE' : 'FREEZE';
+  const modalState = getModalState(data?.account, submitAction, defaultState);
+  const isUnfreezing = modalState === 'UNFREEZE';
+  const contributionsSummary = getRecurringContributionsCountSummary(data?.account);
+  return (
+    <StyledModal maxWidth={600} trapFocus onClose={onClose} {...props}>
+      <CollectiveModalHeader
+        mb={3}
+        collective={collective}
+        preText={
+          isUnfreezing
+            ? intl.formatMessage({ defaultMessage: `Unfreeze`, id: '5SBeLS' })
+            : intl.formatMessage({ defaultMessage: `Freeze`, id: 'Account.Freeze' })
+        }
+      />
+
+      <ModalBody mb={0}>
+        {loading ? (
+          <Skeleton className="h-40" />
+        ) : error ? (
+          <MessageBoxGraphqlError error={error} mb={2} />
+        ) : isUnfreezing ? (
+          <div>
+            <h2 className="mb-4 text-xl font-bold">
+              <FormattedMessage defaultMessage="Are you sure want to unfreeze this collective?" id="OX8+5o" />
+            </h2>
+
+            <p className="mb-6 text-sm">
+              <FormattedMessage
+                defaultMessage="Unfreezing the collective means they will regain full access to the platform."
+                id="dMaivT"
+              />
+              <br />
+              <br />
+              <FormattedMessage
+                defaultMessage="This collective (and all its related Projects & Events) will now have access to accept funds, pay out expenses, post updates, create new Events or Projects."
+                id="1Whmi8"
+              />{' '}
+              <FormattedMessage defaultMessage="Their existing recurring contributions will be resumed." id="VxS/8s" />
+            </p>
+            <Collapsible open={useMessageForAccountAdmins}>
+              <CollapsibleTrigger asChild>
+                <div className="mb-4 flex items-center justify-between rounded-md border border-gray-200 p-4">
+                  <span className="text-sm font-medium text-gray-700">
```

Comment:
```suggestion
                  <span className="text-sm font-medium text-muted-foreground">
```

Reply:
Thanks! I need to force myself to use these `muted` classes rather than `gray`, old habits...

Reply:
Rolling back these ones: `text-gray-700` actually differs from `text-muted-foreground` and the intended design.

Reply:
Muted (looks disabled):
![image](https://github.com/user-attachments/assets/b9c2ead1-02e9-4386-a451-30299d9d82ef)

Gray 700:
![image](https://github.com/user-attachments/assets/4bf88b08-60de-4cf2-b202-72505cae3561)

Moving it to the default back instead:
![image](https://github.com/user-attachments/assets/d926d074-f85e-481a-9b98-0ddaa8dc57e1)

Reply:
@Betree ah, I see! 

I think the default text color applied through shadcn/ui in `app.css` might work in this case, which is `slate-900` (but no need to apply it since it's applied to the body). 

If we want to deviate from that to the `700` variant, we should use `text-slate-700` as this is the color of tailwind grays used by our shadcn/ui implementation.

Reply:
But I see no need to fix this if `gray-700` is used here, we can tackle that in one go for many places and also make it clearer in the tailwind config what we want to use (either remove the `gray` variant or just remap `gray` to the `slate` colors)